analysis/tests/cross_partition_combiners_test.py

```diff
@@ -133,6 +133,16 @@ def test_per_partition_to_cross_partition_utility_only_partition_selection(
         mock_create_for_public_partitions.assert_not_called()
         mock_create_for_private_partitions.assert_called_once_with(0.5)
 
+    def test_sum_metrics_to_data_dropped(self):
```

Comment:
I think it would be beneficial to add a couple test cases here, such as "public partition mode (i.e. partition_keep_probability=1.0" or "empty public partition"